### PR TITLE
Imperial auto units

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Closes #[the issue number this PR is related to]

#### Release notes

Changelog Category: Added | Changed | Deprecated | Removed | Fixed | Security

### CODE REVIEW

File: app/services/variant_units/option_value_namer.rb [hunk 1]
```diff
@@ -64,36 +64,42 @@ def option_value_value_unit_scaled
...
-          1.0 =>         { 'name' => 'g',  'system' => 'metric' },
-          28.35 =>       { 'name' => 'oz', 'system' => 'imperial' },
-          453.6 =>       { 'name' => 'lb', 'system' => 'imperial' },
-          1000.0 =>      { 'name' => 'kg', 'system' => 'metric' },
-          1_000_000.0 => { 'name' => 'T',  'system' => 'metric' }
+          1.0 => { 'name' => 'g', 'system' => 'metric' },
+          28.35 => { 'name' => 'oz', 'system' => 'imperial' },
+          453.6 => { 'name' => 'lb', 'system' => 'imperial' },
+          1000.0 => { 'name' => 'kg', 'system' => 'metric' },
+          1_000_000.0 => { 'name' => 'T', 'system' => 'metric' }
```
Comment (andrewpbrett) [inline app/services/variant_units/option_value_namer.rb:71]: Ah, okay, yeah - I was confused as to what rubocop wanted here, so I aligned the keys.... imo it's easier to read when the keys are lined up, rather than with the whitespace removed, but I can see both sides, so 👍

File: app/services/variant_units/option_value_namer.rb [hunk 2]
```diff
@@ -64,36 +64,42 @@ def option_value_value_unit_scaled
...
+    # If there is none available where this is true, use the smallest available unit.
+    def find_largest_unit(scales, product_scale_system)
       largest_unit = scales.select { |scale, unit_info|
         unit_info['system'] == product_scale_system &&
           @variant.unit_value / scale >= 1
       }.max
-      largest_unit = units[@variant.product.variant_unit].first if largest_unit.nil?
+      return scales.first if largest_unit.nil?
 
-      [largest_unit[0], largest_unit[1]["name"]]
```
Comment (andrewpbrett) [inline app/services/variant_units/option_value_namer.rb:92]: Good idea to pull this into a method. I noticed too that doing things like `largest_unit[1]["name"]` isn't great and would definitely like to refactor further, but can do that in another PR.

File: app/services/variant_units/variant_and_line_item_naming.rb [hunk 3]
```diff
@@ -24,17 +24,20 @@ def options_text
...
+      return option_value.presentation unless option_value.option_type.name == "unit_weight"
+
+      return display_as if has_attribute?(:display_as) && display_as.present?
+
+      return variant.display_as if variant_display_as?
+
       option_value.presentation
     end
 
+    def variant_display_as?
```
Comment (andrewpbrett) [inline app/services/variant_units/variant_and_line_item_naming.rb:36]: Nice 👍

File: app/services/variant_units/option_value_namer.rb (hunk 2 above)
Comment (luisramos0) [reply app/services/variant_units/option_value_namer.rb:92]: yeah, that's how max works. maybe we can add a comment above this line "max returns an array with two elements"

File: app/services/variant_units/option_value_namer.rb (hunk 1 above)
Comment (luisramos0) [reply app/services/variant_units/option_value_namer.rb:71]: ah, yes, looks better aligned. I just wanted to fix rubocop, we can align again and add a rubocop exception for example.
